modules/apps/redirect/redirect-service/src/main/java/com/liferay/redirect/internal/provider/CrawlerUserAgentsProviderImpl.java

```diff
@@ -49,17 +53,26 @@ public boolean isCrawlerUserAgent(String userAgent) {
 			return false;
 		}
 
-		return _crawlerUserAgentsMap.computeIfAbsent(
-			userAgent, this::_isCrawlerUserAgent);
+		Boolean crawlerUserAgent = _portalCache.get(userAgent);
+
+		if (crawlerUserAgent != null) {
+			return crawlerUserAgent;
+		}
+
+		crawlerUserAgent = _isCrawlerUserAgent(userAgent);
+
+		_portalCache.put(userAgent, crawlerUserAgent);
+
+		return crawlerUserAgent;
```

Comment:
@AliciaGarciaGarcia I still see this as calling for trouble. I understand the performance issues, but using user input as cache key is not a good idea. With this code, it is trivially easy to create a one line script that will exhaust all server memory in no time.